genesyscloud/provider.go

```diff
@@ -22,376 +22,455 @@ var resourceMapMutex = sync.RWMutex{}
 }
 
 func getRegionBasePath(region string) string {
-	return "https://api." + getRegionDomain(region)
+        return "https://api." + getRegionDomain(region)
 }
 
 func initClientConfig(data *schema.ResourceData, version string, config *platformclientv2.Configuration) diag.Diagnostics {
-	accessToken := data.Get("access_token").(string)
-	oauthclientID := data.Get("oauthclient_id").(string)
-	oauthclientSecret := data.Get("oauthclient_secret").(string)
-	basePath := getRegionBasePath(data.Get("aws_region").(string))
-
-	config.BasePath = basePath
-	if data.Get("sdk_debug").(bool) {
-		config.LoggingConfiguration = &platformclientv2.LoggingConfiguration{
-			LogLevel:        platformclientv2.LTrace,
-			LogRequestBody:  true,
-			LogResponseBody: true,
-		}
-		config.LoggingConfiguration.SetLogToConsole(false)
-		config.LoggingConfiguration.SetLogFormat(platformclientv2.Text)
-		config.LoggingConfiguration.SetLogFilePath("sdk_debug.log")
-	}
-	config.AddDefaultHeader("User-Agent", "GC Terraform Provider/"+version)
-	config.RetryConfiguration = &platformclientv2.RetryConfiguration{
-		RetryWaitMin: time.Second * 1,
-		RetryWaitMax: time.Second * 30,
-		RetryMax:     20,
-		RequestLogHook: func(request *http.Request, count int) {
-			if count > 0 && request != nil {
-				log.Printf("Retry #%d for %s %s", count, request.Method, request.URL)
-			}
-		},
-		ResponseLogHook: func(response *http.Response) {
-			if response.StatusCode < http.StatusOK || response.StatusCode >= http.StatusMultipleChoices {
-				log.Printf("Response %s", response.Status)
-			}
-		},
-	}
-
-	if accessToken != "" {
-		log.Print("Setting access token set on configuration instance.")
-		config.AccessToken = accessToken
-	} else {
-		err := config.AuthorizeClientCredentials(oauthclientID, oauthclientSecret)
-		if err != nil {
-			return diag.Errorf("Failed to authorize Genesys Cloud client credentials: %v", err)
-		}
-	}
-
-	log.Printf("Initialized Go SDK Client. Debug=%t", data.Get("sdk_debug").(bool))
-	return nil
-}
+        accessToken := data.Get("access_token").(string)
+        oauthclientID := data.Get("oauthclient_id").(string)
+        oauthclientSecret := data.Get("oauthclient_secret").(string)
+        basePath := getRegionBasePath(data.Get("aws_region").(string))
+
+        config.BasePath = basePath
+        if data.Get("sdk_debug").(bool) {
+                config.LoggingConfiguration = &platformclientv2.LoggingConfiguration{
+                        LogLevel:        platformclientv2.LTrace,
+                        LogRequestBody:  true,
+                        LogResponseBody: true,
+                }
+                config.LoggingConfiguration.SetLogToConsole(false)
+                config.LoggingConfiguration.SetLogFormat(platformclientv2.Text)
+                config.LoggingConfiguration.SetLogFilePath("sdk_debug.log")
+        }
+
+        proxySet := data.Get("proxy").(*schema.Set)
+        for _, proxyObj := range proxySet.List() {
```

Comment:
Why is this a list?  Arent proxy objects a single item

Reply:
The list used here is to just iterate and fetch the first item. 
The schema for the proxy is designed as a set of Values. There were issues setting up the resource schema with a Map, hence used this. 
The Schema has been configured with this option 'MaxItems: 1'  which will allow the user only to have one Proxy configuration in his client file.

Reply:
Excellent